pkg/util/purge/resourcegroups.go

```diff
@@ -73,21 +76,43 @@ func (rc *ResourceCleaner) cleanNetworking(ctx context.Context, resourceGroup mg
 		}
 
 		for _, secGroupSubnet := range secGroup.Properties.Subnets {
-			subnet, err := rc.subnet.Get(ctx, *secGroupSubnet.ID)
+			if secGroupSubnet.ID == nil || secGroupSubnet.Name == nil {
+				continue
+			}
+
+			vnetID, _, err := apisubnet.Split(*secGroupSubnet.ID)
 			if err != nil {
 				return err
 			}
 
-			rc.log.Debugf("Removing security group from subnet: %s/%s/%s", *resourceGroup.Name, *secGroup.Name, *subnet.Name)
+			r, err := azure.ParseResourceID(vnetID)
+			if err != nil {
+				return err
+			}
+
+			subnet, err := rc.subnet.Get(ctx, *resourceGroup.Name, r.ResourceName, *secGroupSubnet.Name, nil)
```

Comment:
You can parse subnet ID like this.
https://github.com/Azure/ARO-RP/blob/88f29dee662395c5a34490030c7e15d3477aec3c/pkg/cluster/ensureendpoints.go#L33-L37